Sources/SwiftSyntax/CustomTraits.swift

```diff
@@ -32,7 +60,35 @@ extension ClassDeclSyntax {
   }
 }
 
-extension EnumDeclSyntax {
+extension ClassDeclSyntax {
+  public var header: DeclGroupHeaderSyntax {
```

Comment:
This ties in to my general comment on whether we need `DeclGroupHeaderSyntax`. I would prefer for `ClassDeclSyntax.header` to be of type `ClassDeclHeaderSyntax` instead of `DeclGroupHeaderSyntax`.